qds_sdk/clusterv2.py

```diff
@@ -0,0 +1,545 @@
+        cloud = Cloud.get_cloud()
+        cloud.create_parser(subparser)
+
+        # cluster info parser
+        ClusterInfoV2.cluster_info_parser(subparser, action)
+
+        # engine config parser
+        Engine.engine_parser(subparser)
+
+    @staticmethod
+    def run(args):
+        parser = ClusterCmdLine.parsers(args[0])
+        arguments = parser.parse_args(args)
+        customer_ssh_key = util._read_file(arguments.customer_ssh_key_file)
+
+        # This will set cluster info and monitoring settings
+        cluster_info = ClusterInfoV2(arguments.label)
+        cluster_info.set_cluster_info(disallow_cluster_termination=arguments.disallow_cluster_termination,
+                                      enable_ganglia_monitoring=arguments.enable_ganglia_monitoring,
+                                      datadog_api_token=arguments.datadog_api_token,
+                                      datadog_app_token=arguments.datadog_app_token,
+                                      node_bootstrap=arguments.node_bootstrap_file,
+                                      master_instance_type=arguments.master_instance_type,
+                                      slave_instance_type=arguments.slave_instance_type,
+                                      min_nodes=arguments.initial_nodes,
+                                      max_nodes=arguments.max_nodes,
+                                      slave_request_type=arguments.slave_request_type,
+                                      fallback_to_ondemand=arguments.fallback_to_ondemand,
+                                      custom_tags=arguments.custom_tags,
+                                      heterogeneous_config=arguments.heterogeneous_config,
+                                      maximum_bid_price_percentage=arguments.maximum_bid_price_percentage,
+                                      timeout_for_request=arguments.timeout_for_request,
+                                      maximum_spot_instance_percentage=arguments.maximum_spot_instance_percentage,
+                                      stable_maximum_bid_price_percentage=arguments.stable_maximum_bid_price_percentage,
+                                      stable_timeout_for_request=arguments.stable_timeout_for_request,
+                                      stable_spot_fallback=arguments.stable_spot_fallback,
+                                      idle_cluster_timeout=arguments.idle_cluster_timeout,
+                                      disk_count=arguments.count,
+                                      disk_type=arguments.disk_type,
+                                      disk_size=arguments.size,
+                                      upscaling_config=arguments.upscaling_config,
+                                      enable_encryption=arguments.encrypted_ephemerals,
+                                      customer_ssh_key=customer_ssh_key)
+
+        #  This will set cloud config settings
+        cloud_config = Cloud.get_cloud()
```

Comment:
Updated the function. I have handled it quite differently. In qubole configuration we give cloud details too , then on the basis of this cloud we will initialise cls.cloud_config. Now this cloud_config can be used anywhere like we use cache_agent. Also if the user want to change cloud config then they can always call get_cloud method by passing cloud details explicitly.